src/main.rs

```diff
@@ -147,10 +156,13 @@ fn run(matches: clap::ArgMatches) -> Result<(), Error> {
     let opt_complexity = matches.value_of("complexity").unwrap();
     let opt_count = matches.value_of("count").unwrap();
     let opt_letters = matches.value_of("letters").unwrap();
-    let opt_alliterate = matches.is_present("alliterate") || matches.is_present("ubuntu");
+    let opt_alliterate = matches.is_present("alliterate")
+        || matches.is_present("ubuntu")
+        || matches.is_present("alliterate-with");
 
     // Optional arguments without defaults.
     let opt_directory = matches.value_of("directory");
+    let opt_alliterate_char = matches.value_of("alliterate-with");
```

Comment:
Let's make this an `Option<char>` given that that's what we're calling it:

```suggestion
    let opt_alliterate_char = matches
        .value_of("alliterate-with")
        .map(|s| s.parse::<char>().ok())
        .flatten();
```

This cleans up some code below.